functests/4_latency/latency.go

```diff
@@ -0,0 +1,236 @@
+		}
+
+		var err error
+		profile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+
+		if profile.Spec.CPU.Isolated == nil {
+			Skip(fmt.Sprintf("Skip the oslat test, the profile %q does not have isolated CPUs", profile.Name))
+		}
+
+		isolatedCpus := cpuset.MustParse(string(*profile.Spec.CPU.Isolated))
+		// we require at least two CPUs to run oslat test, because one CPU should be used to run the main oslat thread
+		// we can not use all isolated CPUs, because if reserved and isolated include all node CPUs, and reserved CPUs
+		// do not calculated into the Allocated, at least part of time of one of isolated CPUs will be used to run
+		// other node containers
+		// at least two isolated CPUs to run oslat + one isolated CPU used by other containers on the node = at least 3 isolated CPUs
+		if isolatedCpus.Size() < 3 {
+			Skip(fmt.Sprintf("Skip the oslat test, the profile %q has less than two isolated CPUs", profile.Name))
+		}
+
+		workerRTNodes, err := nodes.GetByLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+		workerRTNodes, err = nodes.MatchingOptionalSelector(workerRTNodes)
+		Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("error looking for the optional selector: %v", err))
+		Expect(workerRTNodes).ToNot(BeEmpty())
+		workerRTNode = &workerRTNodes[0]
+	})
+
+	AfterEach(func() {
+		if err := testclient.Client.Delete(context.TODO(), oslatPod); err != nil {
+			klog.Error(err)
+		}
+	})
+
+	Context("with the oslat image", func() {
+		It("should succeed", func() {
+			oslatPod = getOslatPod(profile, workerRTNode)
+			err := testclient.Client.Create(context.TODO(), oslatPod)
+			Expect(err).ToNot(HaveOccurred())
+
+			timeout, err := strconv.Atoi(latencyTestRuntime)
+			Expect(err).ToNot(HaveOccurred())
+
+			// give two minutes to download the oslat image
+			err = pods.WaitForPhase(oslatPod, corev1.PodRunning, 2*time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+
+			// add additional 2 minutes to give enough time to the cluster to move the pod to Succeeded phase
+			podTimeout := time.Duration(timeout + 120)
+			err = pods.WaitForPhase(oslatPod, corev1.PodSucceeded, podTimeout*time.Second)
+			Expect(err).ToNot(HaveOccurred())
+
+			cmd := []string{"cat", "/rootfs/var/log/oslat.log"}
+			out, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
+			Expect(err).ToNot(HaveOccurred())
+
+			// verify the maximum latency only when it requested, because this value can be very different
+			// on different systems
+			if maximumLatency == -1 {
+				return
+			}
+
+			maximumRegex, err := regexp.Compile(`Maximum:\t*\s*(.*)\s*\(us\)`)
+			Expect(err).ToNot(HaveOccurred())
+
+			latencies := maximumRegex.FindSubmatch([]byte(out))
+			Expect(maximumLatency).ToNot(BeNil())
+
+			// under the output of the oslat very often we have one anomaly high value, for example
+			// Maximum:    16543 15 15 14 13 12 12 13 12 12 12 12 12 12 12 12 12 (us)
```

Comment:
is this a future FIXME ? or skip we need do ?  maybe we need a skip threshold here ?

Reply:
Its nothing we can do under our test, it probably some issue with the oslat and kernel, at least we know that the anomaly value appears only once under the output.